packages/web-config-server/src/apiV1/dataBuilders/helpers/calculateOperationForAnalytics.js

```diff
@@ -41,20 +41,72 @@ const performSingleAnalyticOperation = (analytics, config) => {
   } else if (filteredAnalytics.length === 0) {
     return NO_DATA_AVAILABLE;
   }
-  return OPERATORS[operator](filteredAnalytics[0].value, config);
+  return OPERATORS[operator](filteredAnalytics[0].value, config, models);
 };
 
-const sumDataValues = (analytics, dataValues, filter = {}) => {
-  let sum; // Keep sum undefined so that if there's no data values then we can distinguish between No data and 0
-  const { organisationUnit: organisationUnitFilter } = filter;
+const filterAnalytics = async (analytics, filter, models) => {
+  const {
+    organisationUnit: organisationUnitFilter,
+    parentType: organisationUnitParentType,
+    parentCode: organisationUnitParentCode,
+    parentExcludeCode: organisationUnitParentExcludeCode,
+  } = filter;
+
+  if (!(organisationUnitFilter || organisationUnitParentType || organisationUnitParentCode))
+    return analytics;
+
+  // if filtering on parent lets pre-build a orgUnit-parent map
+  const organisationUnitParentMap = await buildOrganisationUnitParentValue(analytics, models);
+
+  return analytics.filter(a => {
+    if (
+      organisationUnitParentCode &&
+      !checkOrganisationUnitParentValue(
+        organisationUnitParentMap,
+        a.organisationUnit,
+        'code',
+        organisationUnitParentCode,
+      )
+    )
+      return false;
+
+    if (
+      organisationUnitParentExcludeCode &&
+      !checkOrganisationUnitParentValue(
+        organisationUnitParentMap,
+        a.organisationUnit,
+        'excludeCode',
+        organisationUnitParentExcludeCode,
+      )
+    )
+      return false;
+
+    if (
+      organisationUnitParentType &&
+      !checkOrganisationUnitParentValue(
+        organisationUnitParentMap,
+        a.organisationUnit,
+        'type',
+        organisationUnitParentType,
+      )
+    )
+      return false;
```

Comment:
```suggestion
    const parentSatisfiesConditions = Object.entries(parent).every(([field, parentFilter]) =>
      checkValueSatisfiesCondition(parent[field], parentFilter),
    );
    if (!parentSatisfiesConditions) {
      return false;
    }
```

You can then delete the whole `checkOrganisationUnitParentValue` function!

Reply:
I don't check for `parent` existence above - if it's `undefined` an error will be thrown. I believe it's a fair assumption since `World` should be the only org unit with no parent